### PR TITLE
New Layers, aux. functions and built-in model Vision Transformer

### PR DESCRIPTION
**Pull request by carloshnpa**
#### Added layers and functions:
* 3D FullyConnected
* Einsun function
* Dropout 1, 2, 3 and 4 dimensions
* Norm 2D
* Norm 3D
* Softmax 3D and 4D function

#### Running model exemple:
* Vision Transformer (pre-trained with MNIST 28px)

### CODE REVIEW

**Comment by tallesmedeiros** [comment]
@navjotk
